django/contrib/postgres/search.py

```diff
@@ -11,6 +13,20 @@
 from django.db.models.expressions import CombinedExpression, register_combinable_fields
 from django.db.models.functions import Cast, Coalesce
 
+_SEARCH_SPEC_CHARS = r"['\0\[\]()|&:*!@<>\\]"
+_spec_chars_re = re.compile(_SEARCH_SPEC_CHARS)
+multiple_spaces_re = re.compile(r"\s{2,}")
+
+
+def normalize_spaces(val: str):
+    """Converts multiple spaces to single and strips from both sides."""
+    return multiple_spaces_re.sub(" ", val.strip()) if val else None
+
+
+def psql_escape(query: str):
+    """Replace unsafe chars with space and convert multiple spaces to single."""
+    return normalize_spaces(_spec_chars_re.sub(" ", query))
+
```

Comment:
```suggestion

def normalize_spaces(val: str):
    """Converts multiple spaces to single and strips from both sides."""
    return re.sub(r"\s{2,}", " ", val.strip()) if val else None


def psql_escape(query: str):
    """Replace unsafe chars with space and convert multiple spaces to single."""
    return normalize_spaces(re.sub(r"['\0\[\]()|&:*!@<>\\]", " ", query))

```
It seems that you don't need to declare variable since you're not reusing them outside their functions.

Also, if you're not using the compiled regex more than one you don't need to compile them.
See https://docs.python.org/3/library/re.html#re.compile